### DISCUSSION TITLE
competitive mapping?

### DISCUSSION THREAD

**Discussion by mcmahon-uw** [wwood/CoverM#64] [Q&A]
Hi there! Not sure if this is an issue or just a discussion... but it's not a problem just a question.

I'm having a hard time figuring out if there is a way to map in a "competitive" sense. That is, where each read is only counted as mapping to one contig / genome. Or is this the default (in which case, is it possible to change it)?  I looked at the minimap2 documentation a little bit but didn't see anything about it. But perhaps I'm using the wrong terminology?

thank you!
trina

**Comment by wwood** [comment]
Hi @mcmahon-uw sorry for the slow reply.

Yes, one is chosen always as the primary alignment. This is a requirement of the SAM format, which is used as the data format between the mapper (e.g. minimap2) and coverm.

It is possible to include secondary alignments, as defined in the SAM spec

>Multiple mapping: The correct placement of a read may be ambiguous, e.g., due to repeats. In this case,
there may be multiple read alignments for the same read. One of these alignments is considered
primary. All the other alignments have the secondary alignment flag set in the SAM records that
represent them. All the SAM records have the same QNAME and the same values for 0x40 and 0x80
flags. Typically the alignment designated primary is the best alignment, but the decision may be
arbitrary. 

You can include all alignments even if they are secondary with the `--include-secondary` flag, which I imagine pushing out soon.

I'm curious to know whether it makes substantial difference though?

**Comment by bglindner** [comment]
Hey Ben,

Along these lines, I was wondering recently: how are the primary alignments selected when you have a short read that has multiple alignments with the same score? Shuffled at random?

I often want to map competitively to ensure my reads find the right reference genome (or MAG) especially amongst decently close relatives (as opposed to independently mapping against each genome and risking over-counting reads). And I see this edge case come up (i.e., same alignment score -- could go to multiple references) when I'm more intimate with the mapping process. Wondering how coverM handles these cases.

Cheers,

Blake
